concent_api/common/tests/test_api_view.py

```diff
@@ -254,15 +246,55 @@ def setUp(self):
             check_time=False,
         )
         self.force_report_computed_task = message.concents.ForceReportComputedTask(
-            report_computed_task=report_computed_task
+            report_computed_task=self.report_computed_task
+        )
+
+        self.force_get_task_result = message.concents.ForceGetTaskResult(
+            report_computed_task=self.report_computed_task
         )
 
+    def test_that_api_view_should_not_rollback_changes_from_first_transaction_when_second_raises_exception(self):
+        with mock.patch(
+            'core.subtask_helpers.get_one_or_none',
+            side_effect=_create_client_mock
+        ) as _update_timed_out_subtask_correct_response_mock_function, \
+            mock.patch(
+            'core.message_handlers.get_one_or_none',
+            side_effect=_create_client_and_raise_http400_error_mock
+        ) as _create_client_and_raise_http400_error_mock_function:
+            self.client.post(
+                reverse('core:send'),
+                data=dump(
+                    self.force_get_task_result,
+                    PROVIDER_PRIVATE_KEY,
+                    CONCENT_PUBLIC_KEY
+                ),
+                content_type='application/octet-stream',
+            )
+        _update_timed_out_subtask_correct_response_mock_function.assert_called()
+        _create_client_and_raise_http400_error_mock_function.assert_called()
+
+        self.assertEqual(Client.objects.count(), 1)
```

Comment:
There is a method `_assert_client_count_is_equal` in `ConcentIntegrationTestCase`.

Reply:
OK, but I would prefer to leave it as it is now. I think `self.assertEqual(Client.objects.count(), 1)` is short, easy to read and understand. Besides, this test extends `TransactionTestCase`, not `ConcentIntegrationTestCase`. I don't think this change will make code better or easier to understand